apm-agent-core/src/main/java/co/elastic/apm/agent/bci/IndyBootstrap.java

```diff
@@ -196,14 +203,40 @@ public static Method getIndyBootstrapMethod() {
     /**
      * Injects the {@code java.lang.IndyBootstrapDispatcher} class into the bootstrap class loader if it wasn't already.
      */
-    private static Class<?> initIndyBootstrap() throws Exception {
+    private static Class<?> initIndyBootstrap(final Logger logger) throws Exception {
+        Class<?> indyBootstrapDispatcherClass;
         try {
-            return Class.forName(INDY_BOOTSTRAP_CLASS_NAME, false, null);
+            indyBootstrapDispatcherClass = Class.forName(INDY_BOOTSTRAP_CLASS_NAME, false, null);
         } catch (ClassNotFoundException e) {
             byte[] bootstrapClass = IOUtils.readToBytes(ClassLoader.getSystemClassLoader().getResourceAsStream(INDY_BOOTSTRAP_RESOURCE));
             ClassInjector.UsingUnsafe.ofBootLoader().injectRaw(Collections.singletonMap(INDY_BOOTSTRAP_CLASS_NAME, bootstrapClass));
+            indyBootstrapDispatcherClass = Class.forName(INDY_BOOTSTRAP_CLASS_NAME, false, null);
+        }
+
+        if (JvmRuntimeInfo.getMajorVersion() >= 9 && JvmRuntimeInfo.isJ9VM()) {
+            try {
+                logger.info("Overriding IndyBootstrapDispatcher class's module to java.base module");
```

Comment:
[minor] make this log message specific to J9 JVM as it won't happen with other JVMs